include/boost/geometry/strategy/spherical/envelope_range.hpp

```diff
@@ -0,0 +1,283 @@
+            >::type;
+
+        using units_t = typename geometry::detail::cs_angular_units<P>::type;
+        using constants = math::detail::constants_on_spheroid<calc_t, units_t>;
+
+        calc_t const c0 = 0;
+        calc_t const pi = constants::half_period();
+
+        calc_t const lon1 = get<0>(p1);
+        calc_t const lat1 = get<1>(p1);
+        calc_t const lon2 = get<0>(p2);
+        calc_t const lat2 = get<1>(p2);
+        calc_t const lat = get<1>(p);
+
+        calc_t const s_lon_diff = math::longitude_distance_signed<units_t>(lon1, lon2);
+        bool const s_vertical = math::equals(s_lon_diff, c0)
+                             || math::equals(s_lon_diff, pi);
+        // Side of vertical segment is 0 for both poles.
+        if (s_vertical)
+        {
+            return 0;
+        }
+
+        // This strategy shouldn't be called in this case but just in case
+        // check if segment starts at a pole
+        if (math::equals(lat, lat1) || math::equals(lat, lat2))
+        {
+            return 0;
+        }
+
+        // -1 is rhs
+        //  1 is lhs
+        if (lat >= c0) // north pole
+        {
+            return s_lon_diff < c0 ? -1 : 1;
+        }
+        else // south pole
+        {
+            return s_lon_diff > c0 ? -1 : 1;
+        }
+    }
+};
+
+
+template <typename Point, typename Range, typename Strategy>
+inline int point_in_range(Point const& point, Range const& range, Strategy const& strategy)
+{
+    typename Strategy::state_type state;
+
+    auto it = boost::begin(range);
+    auto const end = boost::end(range);
+    for (auto previous = it++ ; it != end ; ++previous, ++it )
+    {
+        if ( ! strategy.apply(point, *previous, *it, state) )
+        {
+            break;
+        }
+    }
+
+    return strategy.result(state);
+}
+
+
+template <typename Ring, typename PoleWithinStrategy>
+inline bool pole_within(bool north_pole, Ring const& ring,
+                        PoleWithinStrategy const& pole_within_strategy)
+{
+    if (boost::size(ring) < core_detail::closure::minimum_ring_size
+                                <
+                                    geometry::closure<Ring>::value
+                                >::value)
+    {
+        return false;
+    }
+
+    using point_t = typename geometry::point_type<Ring>::type;
+    using coord_t = typename geometry::coordinate_type<point_t>::type;
+    using units_t = typename geometry::detail::cs_angular_units<point_t>::type;
+    using constants_t = math::detail::constants_on_spheroid<coord_t, units_t>;
+    point_t point;
+    geometry::assign_zero(point);
+    if (north_pole)
+    {
+        geometry::set<1>(point, constants_t::max_latitude());
+    }
+    else
+    {
+        geometry::set<1>(point, constants_t::min_latitude());
+    }
+    geometry::detail::closed_clockwise_view<Ring const> view(ring);
+    return point_in_range(point, view, pole_within_strategy) > 0;
+}
+
+template
+<
+    typename Range, typename Box,
```

Comment:
According to our guidelines, long template parameter list shall be formatted with one template parameter per line.